obspy/io/mseed/tests/test_mseed_reading_and_writing.py

```diff
@@ -155,6 +155,25 @@ def test_read_and_write_traces(self):
                     np.testing.assert_array_equal(this_stream[0].data,
                                                   new_stream[0].data)
 
+    def test_downcast_int64_to_int32(self):
+        """
+        Checks that sample stream of dtype int64 can be downcast to int 32 and written to mseed
+        by downcasting the data to int32 type data. 
+        """
+        # create a dummy stream with int64 data 
+        x = np.arange(20, dtype = np.int64)
+        tr = Trace(x)
+        st = Stream()
+        st.append(tr)
+
+        # make sure the data can be written to mseed -> write to io.BytesIO() object in test 
+        file = io.BytesIO()
+        st.write(file, format="mseed")
+        st2 = read(file)
```

Comment:
Please use `with` construct, that's automatically closing the buffer at the end..

```
with io.BytesIO() as buf:
    st.write(buf, format="mseed")
    st2 = read(buf)
```

Also `file` is a  Python builtin, try to avoid (locally) overwriting builtins at all cost.

Reply:
Please als otest the failing case as mentioned with `self.assertRaises(...)` but I guess you're still working on this ... :-)